### PR TITLE
chore/v8 tests events

### PR DESCRIPTION
**Pull request by achamas-playco**
### Summary
🔄🧪🖥️

This pull request introduces several changes to the event system and the renderer modules of the pixijs library. It adds a new property to store the initial renderer options, changes the default event mode to 'auto' for better performance and compatibility, adds unit tests for the EventBoundary class, and simplifies the getRenderer function for testing. These changes aim to enhance the functionality, usability, and quality of the pixijs rendering and interaction features.

> _We are the masters of the event system_
> _We unleash the power of `auto` mode_
> _We test the boundaries of every display_
> _We render with options that explode_

### Walkthrough
* Change default event mode from 'passive' to 'auto' for EventSystem and FederatedContainer classes to improve performance and compatibility ([link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-8de4284383bc6a5cb52d3ef6592b1129816a8712991683451bb10849f27eb66fL249-R249), [link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-0a034f50941fd9ad6954e3fdd61cdf7db3a467b4b9fef26e63dd4ff43a6743d0L568-R568))
* Add _initOptions property to AbstractRenderer class to store original options object and assign it in init method ([link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-27fa29afe053bf0bbdb6c29016c4da0abb00506b5b31c3e689aa84f35867aa42R110-R111), [link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-27fa29afe053bf0bbdb6c29016c4da0abb00506b5b31c3e689aa84f35867aa42R156-R158))
* Modify getRenderer function in `tests/utils/getRenderer.ts` to accept partial RendererOptions object and use nullish coalescing and spread operators ([link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-d00f37129bb405183cceab920374886d692196ad4935339a39855693598ffd98L3-R12))
* Add unit tests for EventBoundary class in `tests/events/EventBoundary.tests.ts` to cover various scenarios and edge cases ([link](https://github.com/pixijs/pixijs/pull/9844/files?diff=unified&w=0#diff-e8cfc7d663a22afa92bacce3e31d2ac0ca7f02af362f929cda05b6e1e72ca072R1-R390))

### CODE REVIEW

File: src/rendering/renderers/shared/system/AbstractRenderer.ts [hunk 1]
```diff
@@ -107,6 +107,8 @@ export class AbstractRenderer<PIPES, OPTIONS extends PixiMixins.RendererOptions,
     public view: ViewSystem;
     public textureGenerator: GenerateTextureSystem;
 
+    protected _initOptions: OPTIONS = {} as OPTIONS;
```
Comment (achamas-playco) [inline src/rendering/renderers/shared/system/AbstractRenderer.ts:110]: Have added this purely for the tests to access as I wasn't sure of another possible way.

Comment (GoodBoyDigital) [reply src/rendering/renderers/shared/system/AbstractRenderer.ts:110]: as we are creating the renderer in the tests, do we not know this externally?

File: src/events/EventSystem.ts [hunk 2]
```diff
@@ -249,7 +249,7 @@ export class EventSystem implements System<EventSystemOptions>
 
         this.setTargetElement(canvas as HTMLCanvasElement);
         this.resolution = resolution;
-        EventSystem._defaultEventMode = options.eventMode ?? 'passive';
+        EventSystem._defaultEventMode = options.eventMode ?? 'auto';
```
Comment (Zyie) [inline src/events/EventSystem.ts:252]: We don't want to change this back to `auto`, `passive` should be the default behaviour for v8

File: tests/events/EventBoundary.tests.ts [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+        container.eventMode = 'none';
+        target.interactive = true;
+        blocker.interactive = true;
+
+        expect(boundary.hitTest(25, 50)).toEqual(null);
+        expect(boundary.hitTest(75, 50)).toEqual(null);
+    });
+
+    it.skip('should hit-test correctly when using multiple interactive types', () =>
```
Comment (Zyie) [inline tests/events/EventBoundary.tests.ts:208]: I will look into this before we merge

**Review by Zyie — CHANGES_REQUESTED** [review]

**Review by Zyie — APPROVED** [review]
